build/lite/docker/pull.go

```diff
@@ -60,27 +63,37 @@ func (d Docker) ImagesPullByList(images []string) {
 }
 
 func (d Docker) ImagePull(image string) error {
+	var (
+		named       reference.Named
+		err         error
+		cli         *client.Client
+		authConfig  types.AuthConfig
+		out         io.ReadCloser
+		encodedJSON []byte
+	)
+	named, err = reference.ParseToNamed(image)
+	if err != nil {
+		logger.Warn("image information parsing failed: %v", err)
+		return err
+	}
 	var ImagePullOptions types.ImagePullOptions
 	ctx := context.Background()
-	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	cli, err = client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
 	if err != nil {
+		logger.Warn("docker client creation failed: %v", err)
 		return err
 	}
-	if d.Username != "" && d.Password != "" {
-		authConfig := types.AuthConfig{
-			Username: d.Username,
-			Password: d.Password,
-		}
-		encodedJSON, err := json.Marshal(authConfig)
-		if err != nil {
-			return err
-		}
-		authStr := base64.URLEncoding.EncodeToString(encodedJSON)
-		ImagePullOptions = types.ImagePullOptions{RegistryAuth: authStr}
+	authConfig, _ = utils.GetDockerAuthInfoFromDocker(named.Domain())
```

Comment:
we still need the error, if error happens, just pass a empty RegistryAuth.

Reply:
The GetDockerAuthInfoFromDocker function handles the error and must return an auth object